rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -96,8 +143,34 @@ void SqliteStorage::write(std::shared_ptr<const rosbag2_storage::SerializedBagMe
             "' has not been created yet! Call 'create_topic' first.");
   }
 
-  write_statement_->bind(message->time_stamp, topic_entry->second, message->serialized_data);
-  write_statement_->execute_and_reset();
+  if (extra_file_) {
+    constexpr uint32_t file_index = 0;  // could be changed to split data into files
+
+    auto &file_handle = data_files_[file_index];
+    std::lock_guard<std::mutex>(file_handle->lock);
+
+    MetaInformation data = {
+      static_cast<uint64_t>(file_handle->file_stream.tellg()),
+      static_cast<uint64_t>(message->serialized_data->buffer_length),
+      file_index
+    };
+
+    file_handle->file_stream.write(reinterpret_cast<const char *>(message->serialized_data->buffer),
+      message->serialized_data->buffer_length);
+
+    auto serialized_data = std::make_shared<rcutils_uint8_array_t>();
```

Comment:
I am having trouble to understand what's happening with this `serialized_data` instance.